### DISCUSSION TITLE
Disable all `threshold` parameters for stable transciption when using `vad_filter=True`.

### DISCUSSION THREAD

**Discussion by zh-plus** [SYSTRAN/faster-whisper#349] [Q&A]
Recently I found that large-v2 model usually drops some transcribed segments for Chinese and Japanese because of `compression_ratio_threshold`, `log_prob_threshold`, and `no_speech_threshold`. However, the outputted transcription is actually correct.

Is it a good practice to totally disable these 3 parameters by:
```python
default_asr_options = {
  "compression_ratio_threshold": None,
  "log_prob_threshold": None,
  "no_speech_threshold": None,
}
```  
when setting `vad_filter=True`? Perhaps it would be more appropriate for a standalone VAD model to determine which voice segments should be disregarded.

The reliability of `log_prob_threshold` is often questioned, as mentioned in this discussion on https://github.com/openai/whisper/discussions/29#discussioncomment-3726710.

**Comment by dgoryeo** [comment]
Hi @zh-plus , I was wondering, does the code above disables the threshold parameters, or does it set them to default?  Thanks for sharing your insights.

**Reply by zh-plus** [reply]
It disables the threshold according to https://github.com/guillaumekln/faster-whisper/blob/7b271da0351e4f81f80e8bb4d2c21c9406475aa9/faster_whisper/transcribe.py#L658

**Comment by dgoryeo** [comment]
Hi @zh-plus , I get an error when i try to set log_prob_threshold to None.  Is there anything I am doing wrong? 

```

    options2 = {"language":language],
                "task":task,
                "word_timestamps":True,
                "temperature":(0.0, 0.2),
                "best_of":5,
                "beam_size":5,
                "patience":2,
                "suppress_tokens":"",
                "initial_prompt":None,
                "condition_on_previous_text":False,
                "compression_ratio_threshold":None,
                "no_speech_threshold":None,
                "log_prob_threshold":None,
                "vad_filter":True,
                "vad_parameters":dict(threshold=vad_threshold, max_speech_duration_s=chunk_duration)
                }

    segments, info = WHISPER_MODEL.transcribe(
      audio_path,
      **options2
    )


```

Error message is: 

`TypeError: generate(): incompatible function arguments.`

Thanks.

**Reply by zh-plus** [reply]
I don't see any issues with the code snippet and error message you posted. Could you please provide the full exception stack?

**Comment by dgoryeo** [comment]
Thanks for looking into it @zh-plus. I didnt save the entire error message I'm afraid. However I turned to GPT4, and "his" recommendation was to change to `"suppress_tokens":[]` . Syntactically it went through with no error but I'm not sure if `[]` would perform the same as `""`.  It seems that the culprit was supress_tokens than log_prob.

**Reply by zh-plus** [reply]
Yes, it's the same. You can check https://github.com/guillaumekln/faster-whisper/blob/7b271da0351e4f81f80e8bb4d2c21c9406475aa9/faster_whisper/transcribe.py#L60C12-L60C12 for all default values.

**Comment by toanhuynhnguyen** [comment]
Does it actually work?
